sympy/solvers/solveset.py

```diff
@@ -487,9 +487,26 @@ def solveset_real(f, symbol):
     elif f.is_Piecewise:
         result = EmptySet()
         expr_set_pairs = f.as_expr_set_pairs()
-        for (expr, in_set) in expr_set_pairs:
-            solns = solveset_real(expr, symbol).intersect(in_set)
-            result = result + solns
+        expr_cond = {}
+        sym_interval ={}
+        for other_n, (expr_tmp, cond) in enumerate(f.args):
```

Comment:
I was looking for other way but it was not working.
Can you suggest me what method I need to use to get only `expr` and `cond` ?